demo/x11_rawfb/nuklear_rawfb.h

```diff
@@ -70,26 +70,50 @@ struct rawfb_context {
 #endif
 
 static unsigned int
-nk_color_from_byte(const nk_byte *c)
+nk_int_from_color(const struct nk_color *c)
 {
     unsigned int res = 0;
 #if defined(RAWFB_RGBX_8888)
```

Comment:
Because `RAWFB_RGBX_8888` and `RAWFB_XRGB_8888` are mutually exclusive, could you extend all these ifdefs to `#if defined(RAWFB_RGBX_8888) && ! defined(RAWFB_XRGB_8888)` on all the places in the code?

Also I would prefer in this case pass by value as the dereferencing will be most probably more costly considering the size of the struct compared to the size of a pointer (because compilers are way more constrained when working with pointers than with copied values). Usually I wouldn't point this out, but this function is called for each pixel, which is definitely the critical path and such simple "optimization" is the lowest hanging fruit :wink:.

Reply:
Done and done. While I'm content leaving the #define macros for RGBX/XRGB for now, the more I think about it, the more I think this shouldn't be a compile-time decision. This is all data that can be pulled from the FBIOGET_VSCREENINFO ioctl. If you don't mind, I'd like to get this PR closed out and I'll make a separate pull request after I get that designed and implemented properly. I'll probably add some things to the rawfb_image struct at init time to track pixel order and have the necessary routines check it on the fly.  I'll also need to check the other demos/back-ends to see if they need similar changes. But again... separate issue?

Reply:
Currently the pull request is clean from my point of view and I would immediately merge it. But if you say you'll do a separate request, then you're welcome and I'm closing this one.

Thanks for your contributions!